Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt

```diff
@@ -0,0 +1,154 @@
+package net.skyscanner.backpack.fab
+
+import android.animation.AnimatorInflater
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.RippleDrawable
+import android.graphics.drawable.ShapeDrawable
+import android.graphics.drawable.shapes.OvalShape
+import android.util.AttributeSet
+import androidx.annotation.ColorInt
+import androidx.appcompat.widget.AppCompatButton
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.DrawableCompat
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.BpkTheme
+import net.skyscanner.backpack.util.createContextThemeWrapper
+import net.skyscanner.backpack.util.darken
+
+internal open class BpkFab @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : AppCompatButton(createContextThemeWrapper(context, attrs, R.attr.bpkFabStyle), attrs, defStyleAttr) {
```

Comment:
Can we not extend from `FloatingActionButton`. Would save a lot of effort and give all the other functionalities of the component.

Reply:
The general idea here is to migrate all (BpkButton, BpkFab) to the same base class and have the same functionality and api -- what's why it extends AppCompatButton.